provider/instance.go

```diff
@@ -0,0 +1,133 @@
+package provider
+
+import (
+	"crypto/sha1"
+	"encoding/hex"
+	"reflect"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/roblox/terraform-provider-maas/maas"
+)
+
+// MACAddress is used by the maas_instance resource
+type MACAddress maas.MACAddress
+
+// Instance models the maas_instance Terraform resource
+type Instance struct {
```

Comment:
I would love to store more of this data in the terraform state. This will be useful for looking up metadata and passing it down to other resources. One use case here is to have Terraform update Netbox after a node has been deployed.